memiavl/db.go

```diff
@@ -97,6 +99,8 @@ type Options struct {
 	// truncate the versions after the `TargetVersion`, the `TargetVersion` becomes the latest version.
 	// it do nothing if the target version is `0`.
 	LoadForOverwriting bool
+
+	SnapshotWriterLimit int
```

Comment:
In fillDefaults, should we reset it to the default value if it's zero or negative?